### ISSUE TITLE
User Spoofing Exploit (?)

### ISSUE THREAD

**Issue by Sceleratis** [Sceleratis/Adonis#227]
I've received a report (ongoing) that exploiters have been somehow running commands as admins. I do not know enough about how they're doing this to know if it's an Adonis exploit or a Roblox exploit (or a backdoor in the game???) but am currently assuming the latter. 

![unknown-1.png](https://user-images.githubusercontent.com/18238620/103835366-17b2c980-5054-11eb-898c-0054424e852f.png)

The user in these logs running commands is apparently not the actual user.

I don't believe there is reason to panic currently due to the limited evidence/impact observed thus far (over the last week or so.)

I'm seeking information on this so I know how to proceed. If you have any info about how they're doing this or if it's happened to you, please post about it here so we can figure it out together. 

In the mean time, if you're worried I'd recommend disabling the SaveAdmins setting (or the datastore stuff entirely) so people can't save new admin users to the datastore (for after this is eventually patched.)

I'd like to stress again that I don't even know if this is Adonis specific. There is a strong possibility that this is yet another user spoofing exploit on Roblox, in which case it will affect much more than just Adonis and can only be fixed by Roblox themselves (assuming they haven't already.)

**Comment by TheLegendarySpark** [comment]
Did they use a different kind of Adonis? What were their admins? The attachment doesn't help, even though; there could have been more background on the remote logs and stuff.

**Comment by Sceleratis** [comment]
> Did they use a different kind of Adonis? What were their admins? The attachment doesn't help, even though; there could have been more background on the remote logs and stuff.

This is all I know so far. As far as I can tell, they seem to be using the current version of Adonis. 

Screenshot was provided by the user. I waited and watched remote logs for a bit in an active server but nothing was happening.

**Comment by ccuser44** [comment]
Are you sure this is not because of of some backdoor.
This could be related to the backdoor which changes the module id.

Also could this be related to the proposed displayname feature in Roblox?

Also check the admin list of the game if the user in the logs is there or has ever been an admin. Also the reason why they could  run commands as a different player could be that they have a backdoor and use some ServerSide chathax script to say messages as differnt players.

**Comment by Mooshua** [comment]
> I'd like to stress again that I don't even know if this is Adonis specific. There is a strong possibility that this is yet another user spoofing exploit on Roblox, in which case it will affect much more than just Adonis and can only be fixed by Roblox themselves (assuming they haven't already.)

I'd say Roblox is secure. If someone found a way to change the player of a FireServer request, we'd figure out a long time before it were abused by being petty with Adonis. Nonetheless, there are currently no such _known_ exploits.

> Also could this be related to the proposed displayname feature in Roblox?

Adonis does not reference `DisplayName`

Overall, I'm not exactly sure what to think of this. The attack is low-scale enough that I'd say it would be an external backdoor (chat hacks, loadstring, etc). Reasoning:

1. Admin privileges were not removed or added
2. Actions showed up in logs
    - Counter: Exploiter wished to frame a user for the attack
3. No ultra-harmful commands were used- implying that the exploiter was limited to the commands that Mors (the victim) could use.

The best way to patch the exploit would probably be an interception of *all* commands at the client stage- and pushing them through a RemoteEvent to the server. This would require a fork of the chat system, or a plugin. (Beware- a plugin might be awfully *un-doable* by any backdoor worth it's cents)

At the same time, I'd advise your friends who submitted this to do a plugin audit of their developer team, and search their server-scripts thoroughly. You never know what you might dig up.

**Comment by Sceleratis** [comment]
I mainly posted this in attempt to see if anyone else faced this or may have any ideas. My main assumption has been a backdoor since the group the user who contacted me belongs to is fairly large and has games that are fairly active with multiple developers (and thus a prime target for pretty much anything.)

The only reason I suspected it might be something related to Roblox was due to previous exploits that behaved identically.

Since then, the user noted they haven't had any new incidences so I'm going to close this.

If anyone runs into this or has a better idea on what was happening, this can be reopened 🤷‍♀️
